src/pydash/objects.py

```diff
@@ -535,7 +905,7 @@ def get(obj, path, default=None):
     return obj
 
 
-def has(obj, path):
+def has(obj: t.Iterable, path: t.Union[str, int, t.List[t.Union[str, int]]]) -> bool:
```

Comment:
`path` should be able to be `path: t.Union[t.Hashable, t.List[t.Hashable]]`.

Similarly for any other `path` argument that's used as dictionary keys.

Reply:
This sounds nice. Also made me realize that having a type alias for this would be nice as it is used in several places.

So I added a `PathT` type alias with the hashable union you suggested and used it where possible